workspaces/libnpmpack/test/index.js

```diff
@@ -172,3 +171,40 @@ t.test('runs scripts in foreground when foregroundScripts === true', async t =>
     process.chdir(cwd)
   })
 })
+
+t.test('doesn\'t run scripts when ignoreScripts === true', async t => {
+  const testDir = t.testdir({
+    'package.json': JSON.stringify({
+      name: 'my-cool-pkg',
+      version: '1.0.0',
+      scripts: {
+        prepack: 'touch prepack',
+      },
+    }, null, 2),
+  })
+
+  const cwd = process.cwd()
+  process.chdir(testDir)
+
+  const [scriptShell, scriptArgs] = makeSpawnArgs({
+    event: 'prepack',
+    path: testDir,
+    cmd: 'touch prepack',
+  })
+
+  const prepack = spawk.spawn(scriptShell, scriptArgs)
+
+  await pack('file:.', {
+    packDestination: testDir,
+    foregroundScripts: true,
+    ignoreScripts: true,
+  })
+
+  t.ok(!prepack.called)
+
+  spawk.clean()
```

Comment:
This is required because spawk doesn't provide any other mechanism to exclude an interceptor from the uncalled check in `spawk.done()`, causing the test to fail. Is this acceptable, or would you rather (me, or you) implement something cleaner in spawk for this?

Reply:
The cleaner way to do this is to put it in the `t.teardown` so that even if the `t.ok` fails, `spawk.clean()` is called.

Reply:
Doesn't look like that works as expected, see test failures.

Reply:
Go ahead and revert it then.  I'm heads down trying to get tests fixed in several repos so we can land one of the last breaking changes for 9 so I don't have time to debug this.  We shouldn't let this block the PR.  Get the tests passing again and we can worry about perfection tomorrow.

Reply:
I won't be able to debug or revert this for at least 3-4 hours from now. If you need to land this ASAP, force push the branch to 613a6e9db16e5edbfd059acdfe0672c766649942.

Reply:
No worries! The cli is on a weekly release cadence so as long as it's ready before tomorrow morning PDT we should be ok.  And if not it's ok too, npm 9 doesn't drop out of pre-release mode for another week at least.

Reply:
I moved `spawk.done()` in tspawk from `afterEach` to `teardown`, which looks like it works as expected.